### ISSUE TITLE
OS X support

### ISSUE THREAD

**Issue by Leandros** [JBakamovic/yavide#20]
Currently OS X is not supported, but most, if not all, should be working on OS X.
Things I noticed in the install script:

- Add `brew` as OS X package manager

```
    elif [[ `which brew` != "" ]]; then
        SYSTEM_PACKAGE_MANAGER="brew"
        SYSTEM_PACKAGE_MANAGER_INSTALL="brew install"
        SYSTEM_PACKAGE_MANAGER_UPDATE="brew update"
    fi
```

- Remove sudo from calls, they're not necessary since everything should be installed in user space, not system space.

**Comment by JBakamovic** [comment]
I am aware that OS X can be supported almost out of the box but I don't have any Mac machine to test the setup. If you can confirm that the change you mention works on your machine I will be happy to apply it.

**Comment by JBakamovic** [comment]
I forgot to comment about removing `sudo` calls. I am afraid this is not possible as there are quite a few system packages that are required to be installed to set-up the `Yavide` environment properly. Installation process of those packages require R/W access to system directories, hence `sudo` access will be needed.

Alternative ways to circumvent the `sudo` requirement would be to have:
* users install number of dependencies by themselves or
* administrators install the dependencies for ordinary users or
* tweak the installation script to build each dependency manually and install it into the `$HOME` environment

I don't find convenient none of these alternatives. Let me know if you had something in mind.

**Comment by Leandros** [comment]
Yes, that's the main issue why I haven't just added the missing steps to the script and submit a pull-request. The only other idea I could come up with, is adding a 2nd script which is solely for installing on os x. I think that might fit quite well, since on os x the user could also use different package managers.

**Comment by JBakamovic** [comment]
Ok, I've been reading a bit about the `homebrew` philosophy and I think I understand your point now. I think that having a second script, basically the same as the first one, would be an overkill and would probably lead to maintenance issues.

One idea that comes up to my mind is to refactor the installation script so that it checks if it has R/W access to the relevant directories and act accordingly. Provided that access is restricted and there is no `sudo` permission available it would result with stopping the installation process.

**Comment by jesuisnicolasdavid** [comment]
Is there any way to make this awesome idea compatible with osx ? I changed the install script with brew, but i have a gvim error of course. Thanks for your help

**Comment by JBakamovic** [comment]
Hi Nicolas, what specific error are you getting? I think that there shouldn't be any bigger problems making this compatible with osx.
